tests/components/mysensors/test_config_flow.py

```diff
@@ -0,0 +1,559 @@
+    CONF_VERSION,
+    DOMAIN,
+    ConfGatewayType,
+)
+from homeassistant.components.mysensors.gateway import is_serial_port
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+
+async def get_form(
+    hass: HomeAssistantType, gatway_type: ConfGatewayType, expected_step_id: str
+):
+    """Get a form for the given gateway type."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    stepuser = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert stepuser["type"] == "form"
+    assert not stepuser["errors"]
+
+    actualstep = await hass.config_entries.flow.async_configure(
+        stepuser["flow_id"],
+        {CONF_GATEWAY_TYPE: gatway_type},
+    )
+    await hass.async_block_till_done()
+    assert actualstep["type"] == "form"
+    assert actualstep["step_id"] == expected_step_id
+
+    return actualstep
+
+
+async def test_config_mqtt(hass: HomeAssistantType):
+    """Test configuring a mqtt gateway."""
+    step = await get_form(hass, CONF_GATEWAY_TYPE_MQTT, "gw_mqtt")
+    flowid = step["flow_id"]
+
+    with patch(
+        "homeassistant.components.mysensors.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.mysensors.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            flowid,
+            {
+                CONF_RETAIN: True,
+                CONF_TOPIC_IN_PREFIX: "bla",
+                CONF_TOPIC_OUT_PREFIX: "blub",
+                CONF_VERSION: "2.4",
+            },
+        )
+        await hass.async_block_till_done()
+
+    if "errors" in result2:
+        assert not result2["errors"]
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "mqtt"
+    assert result2["data"] == {
+        CONF_DEVICE: "mqtt",
+        CONF_RETAIN: True,
+        CONF_TOPIC_IN_PREFIX: "bla",
+        CONF_TOPIC_OUT_PREFIX: "blub",
+        CONF_VERSION: "2.4",
+    }
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+def test_is_serial_port_windows(hass: HomeAssistantType):
+    """Test windows serial ports."""
+    tests = {
+        "COM5": True,
+        "asdf": False,
+        "COM17": True,
+        "COM": False,
+        "/dev/ttyACM0": False,
+    }
+
+    def testport(port, result):
+        try:
+            is_serial_port(port)
+        except vol.Invalid:
+            success = False
+        else:
+            success = True
+        assert success == result
+
+    with patch("sys.platform", "win32"):
+        for test, result in tests.items():
+            testport(test, result)
+
+
+async def test_config_serial(hass: HomeAssistantType):
+    """Test configuring a gateway via serial."""
+    step = await get_form(hass, CONF_GATEWAY_TYPE_SERIAL, "gw_serial")
+    flowid = step["flow_id"]
```

Comment:
Please separate words with underscore in variable names.
```suggestion
    flow_id = step["flow_id"]
```